src/main/java/net/neoforged/neoforge/transfer/transaction/TransactionManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.transaction;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Manager for handling opening new {@link Transaction Transactions} or querying status of a transaction chain in a given thread.
+ * 
+ * @see Transaction
+ */
+public final class TransactionManager {
```

Comment:
Keep these methods in the `Transaction` interface, there was no need to move them out.

Reply:
I would prefer to have the methods to open a Transaction in the Transaction class rather than moved to another class, so that everything is kept in one place.

Reply:
I may be late, but is there any particular reason why Transaction cannot just be a final class? If you want `Transaction.open` as an API, there should really only ever be a single implementation of it. Having an additional interface seems like designing for a scenario that will by design never happen.